Hazel.UnitTests/TestHelper.cs

```diff
@@ -270,6 +270,37 @@ internal static void RunClientDisconnectTest(NetworkConnectionListener listener,
             mutex2.WaitOne();
         }
 
+        /// <summary>
+        ///     Ensures a client sends a disconnect packet to the server on Dispose.
+        /// </summary>
+        /// <param name="listener">The listener to test.</param>
+        /// <param name="connection">The connection to test.</param>
+        internal static void RunClientDisconnectOnDisposeTest(NetworkConnectionListener listener, Connection connection)
+        {
+            ManualResetEvent mutex = new ManualResetEvent(false);
+            ManualResetEvent mutex2 = new ManualResetEvent(false);
+
+            listener.NewConnection += delegate (NewConnectionEventArgs args)
+            {
+                args.Connection.Disconnected += delegate (object sender2, DisconnectedEventArgs args2)
+                {
+                    mutex2.Set();
+                };
+
+                mutex.Set();
+            };
+
+            listener.Start();
+
+            connection.Connect();
+
+            mutex.WaitOne();
```

Comment:
Suggest a timeout and Assert.Fail in case we break this test, so it doesn't never finish.